### ISSUE TITLE
Lua: pcall does not work if wraps an asynchronous function

### ISSUE THREAD

**Issue by immerrr** [scrapinghub/splash#165]
The reproducible example is as follows:
```
from splash.tests.stress import lua_runonce
lua_runonce("""
function main(splash)
print(pcall(function()
    print("Hello, world")
    splash:wait(0.1)
    print("Hello, world #2")
end))
end
""")
```

It ends up printing both statements on the screen, but ends with:
```
Traceback (most recent call last):
  File "../../test_pcall.py", line 12, in <module>
    """)
  File "/home/immerrr/splash/splash/tests/stress.py", line 145, in lua_runonce
    raise RuntimeError(resp.text)
RuntimeError: unhandled Lua error: error in error handling
```

Interaction between `pcall` & `coroutine.yield` has been an issue for quite a while, but it is solved in lua5.2. Maybe lupa's handling of coroutines has something to do with this.

Apparently, this issue is a blocker for #161.

**Comment by immerrr** [comment]
It appears to be a bug in Lupa.  I've reported it upsteam as scoder/lupa#48.

**Comment by kmike** [comment]
Hmm, when I submit 

```lua
function main(splash)
print(pcall(function()
    print("Hello, world")
    splash:wait(0.1)
    print("Hello, world #2")
end))
end
```

to Splash it prints

```
Hello, world
Hello, world #2
true
```
as expected, without any error messages. I've tried it in Splash master in OS X and with our Docker image.

**Comment by kmike** [comment]
Aha, it fails with scrapinghub/splash:1.8 but passes with scrapinghub/splash:master. It looks like a Lua bug fixed by upgrading to Ubuntu 14.04: 12.04 had Lua 5.2.0, 14.04 has 5.2.3.

**Comment by kmike** [comment]
@immerrr I can't say I understand https://github.com/scoder/lupa/issues/48 in full though. Does it still affect Splash, could you please check that?

**Comment by immerrr** [comment]
This is indeed a problem with Lua 5.2.0 which is fixed in 5.2.1: http://www.lua.org/bugs.html#5.2.0-4.
